### PR TITLE
MAP integration tests

### LINKED ISSUE #5882: Migration of Cypress tests -> Integration tests
https://paper.dropbox.com/doc/Simorgh-Integration-tests-for-pages--AwPtVXQFVXVt2aRxU7R5F64xAg-dj7GKgjKZGWNMxPOpgxyp

### LINKED ISSUE #5917: Migrate MAP tests from Cypress -> React-Testing-Library

### PR DESCRIPTION
**Pull request by karinathomasbbc**
Resolves #5917 

**Overall change:** 
Integration Tests for MAP pages - Pidgin (CPS), Persian (CPS & TC2)

**Code changes:**
- Adding tests
- Updating article tests which correct URL

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] I have added labels to this PR for the relevant pod(s) affected by these changes
- [x] I have assigned this PR to the Simorgh project

**Testing:**
- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [x] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`) 
- [ ] This PR requires manual testing

### CODE REVIEW

File: src/integration/utilities/escapeHTMLText/index.js
Comment (jroebu14) [inline src/integration/utilities/escapeHTMLText/index.js]: what does this do?

**Review by jroebu14 — COMMENTED** [review]
This looks great!

My only comments would be about the abstractions like `getHeadlineElement`. I prefer to keep things explicit rather than make abstractions that don't abstract all that much. This goes against the DRY principle but I think we should "prefer duplication over the wrong abstraction" as discussed here https://www.sandimetz.com/blog/2016/1/20/the-wrong-abstraction

I also don't think we shouldn't put assertions into utilities like `shouldRenderHeadline` and keep all assertions together

**Comment by jroebu14** [comment]
Keen to merge this in after discussing my points above. Thanks!

File: src/integration/utilities/escapeHTMLText/index.js
Comment (karinathomasbbc) [reply src/integration/utilities/escapeHTMLText/index.js]: It is no longer needed - deleted

**Comment by karinathomasbbc** [comment]
> This looks great!
> 
> My only comments would be about the abstractions like `getHeadlineElement`. I prefer to keep things explicit rather than make abstractions that don't abstract all that much. This goes against the DRY principle but I think we should "prefer duplication over the wrong abstraction" as discussed here https://www.sandimetz.com/blog/2016/1/20/the-wrong-abstraction
> 
> I also don't think we shouldn't put assertions into utilities like `shouldRenderHeadline` and keep all assertions together

Thanks for this. I have updated the PR in line with this, hopefully it's a lot more explicit. Definitely a fan of duplication if it means we have clarity & readability.

File: src/integration/pages/mediaAssetPage/user.js
Comment (karinathomasbbc) [inline src/integration/pages/mediaAssetPage/user.js]: If you unskip this, you will see why it fails - might need some assistance on how to fix it.

File: src/integration/common/a11y.js
Comment (jroebu14) [inline src/integration/common/a11y.js]: I don't think this selector needs to change because we have an assertion that explicitly checks for an id of content on line 16

Comment (karinathomasbbc) [reply src/integration/common/a11y.js]: The MAP a11y tests fail because there are 2 elements with the same text in the HTML - the VisuallyHiddenText & StyledFauxHeadline. Unless we fix the underlying issue (which I believe you had raised here: https://github.com/bbc/simorgh/issues/5688 but it has since been closed), then this won't work.

File: package.json
Comment (karinathomasbbc) [inline package.json]: This should enable integration tests in CI

File: src/integration/pages/mediaAssetPage/user.js
Comment (jroebu14) [inline src/integration/pages/mediaAssetPage/user.js]: Is `within` use here necessary? Also, that's a weak looking selector. I can imagine other elements in the page could have a class of Wrapper. I wouldn't encourage class name query selectors anyway especially in the user tests. A stronger query selector might be something like `main [role="list"]`. Even then you could use one of the DOM Testing Library queries for this sort of thing https://testing-library.com/docs/dom-testing-library/api-queries such as `getByRole` e.g.

```js
const [bulletedListItemEl] = page.getByRole('list')
```

Comment (karinathomasbbc) [reply src/integration/pages/mediaAssetPage/user.js]: This scenario - and the related content one - really need some work.  They both use lists, and sometimes it is not possible to determine which one is which, and that's why I had to revert to looking at the class.  It think we should take a look at these in more detail - once the rest has been merged.

File: package.json
Comment (karinathomasbbc) [inline package.json]: This is a reminder that we need to re-enable the dependencies check before merging to latest

File: package.json
Comment (karinathomasbbc) [inline package.json]: Integration tests run locally

File: package.json
Comment (karinathomasbbc) [inline package.json]: Integration tests run in CI before any of the browser tests. Meaning they fail fast if there are any issues.

File: package.json
Comment (karinathomasbbc) [inline package.json:34]: I couldn't get this to play ball by setting `testPathIgnorePatterns` in the `jest.config.js` file, so this was the (hacky) workaround

**Review by karinathomasbbc — COMMENTED** [review]
.

File: package.json
Comment (karinathomasbbc) [reply package.json]: I have fixed it - no action required

File: src/integration/pages/mediaAssetPage/user.js
Comment (karinathomasbbc) [inline src/integration/pages/mediaAssetPage/user.js]: I need a second opinion on this - not sure if we should do this or not.

File: src/integration/pages/articlePage/pidgin.test.js
Comment (karinathomasbbc) [inline src/integration/pages/articlePage/pidgin.test.js:2]: NB: the previous pathname `/pidgin/tori-51745682` was an STY rather than an article

Comment (jroebu14) [reply src/integration/pages/articlePage/pidgin.test.js:2]: Thanks for fixing

File: src/integration/pages/mediaAssetPage/user.amp.js
Comment (jroebu14) [inline src/integration/pages/mediaAssetPage/user.amp.js]: is this necessary?

File: src/integration/pages/mediaAssetPage/user.js
Comment (jroebu14) [reply src/integration/pages/mediaAssetPage/user.js]: You mean the skip? Yeh I'm not sure about that. Nice idea but we're using a feature in a way that it's not intended to be used

File: src/integration/pages/mediaAssetPage/user.js
Comment (jroebu14) [inline src/integration/pages/mediaAssetPage/user.js]: Is it possible to not have a class selector especially with the `^=` as I need to check what it does every time I see it. Not sure we should rely on class selectors in these tests. I think the headline text for related content will be unique anyway

File: src/integration/pages/mediaAssetPage/user.js
Comment (karinathomasbbc) [reply src/integration/pages/mediaAssetPage/user.js]: Removed

File: src/integration/pages/mediaAssetPage/user.amp.js
Comment (karinathomasbbc) [reply src/integration/pages/mediaAssetPage/user.amp.js]: Removed 👍

File: src/integration/pages/mediaAssetPage/user.js
Comment (karinathomasbbc) [reply src/integration/pages/mediaAssetPage/user.js]: I have rewritten this to avoid using class selectors

File: src/integration/pages/mediaAssetPage/user.js
Comment (jroebu14) [inline src/integration/pages/mediaAssetPage/user.js]: should the if block be outside of the it block?

**Review by jroebu14 — APPROVED** [review]

File: src/integration/common/a11y.js
Comment (ryanmccombe) [inline src/integration/common/a11y.js:4]: 🧐 where is `amp` and `canonical` defined?

**Review by ryanmccombe — APPROVED** [review]
It's really difficult to effectively review big sweeping PRs like this.

I think the best strategy for these testing PRs is probably to just merge them all - we can review them later (and iterate if needed) once everyone has worked with the new setup and is more familiar

Would benefit from some higher-level documentation once all these PRs are merged and integrated

Thanks for taking on this massive chunk of work! 👍

**Comment by karinathomasbbc** [comment]
> It's really difficult to effectively review big sweeping PRs like this.
> 
> I think the best strategy for these testing PRs is probably to just merge them all - we can review them later (and iterate if needed) once everyone has worked with the new setup and is more familiar
> 
> Would benefit from some higher-level documentation once all these PRs are merged and integrated
> 
> Thanks for taking on this massive chunk of work! 👍

Thanks for reviewing - it's sometimes tricky to make these PRs small because they can involve sweeping changes. Jonathan has already written some really good docs on this - https://github.com/bbc/simorgh/blob/latest/src/integration/README.md, but these may need to be updated once the newer set of changes are merged (autogenerate tests for 41 services).

**Pull request merged**
